src/pretix/base/services/cart.py

```diff
@@ -1416,14 +1443,11 @@ def recompute_final_prices_and_taxes(self):
     def commit(self):
         self._check_presale_dates()
         self._check_max_cart_size()
-        self._calculate_expiry()
 
         err = self._delete_out_of_timeframe()
         err = self.extend_expired_positions() or err
         err = err or self._check_min_per_voucher()
 
-        self.real_now_dt = now()
```

Comment:
Not really scope of this PR but since you touched it: This was there because when `_extend_expiry_of_valid_existing_positions`, it is *technically* possible that it validates cart positions that are not actually valid any more, because whenever  this code executes, `real_now_dt` is already in the past. So this way my paranoid try to make this error scenario less likely. Being aware there is no chance to fully avoid the error scenario without going deep into clock synchronization (and transaction isolation)… should we maybe just add a small timedelta (eg 5s) to the `expires__gt=self.real_now_dt` to avoid the risk, i.e. only extending the expiry if the current one is still a few seconds in the future? Selling too many tickets is always worse than to little…